openrtx/src/rtx/OpMode_M17.cpp

```diff
@@ -335,6 +349,28 @@ void OpMode_M17::txState(rtxStatus_t *const status)
         modulator.sendFrame(m17Frame);
     }
 
+    gps_t gps_data;
+    gps_data = state.gps_data;
+    if(encoder.canUpdateLsf() && state.settings.gps_enabled && (gpsTimer == -1 || gpsTimer >= MINIMUM_GPS_TIME_TICKS) && gps_data.fix_type > FIX_TYPE_NOT_AVAIL)
+    {
+        gpsTimer = 0;
+        gpsTransmitting = true;
+        lsf.setGnssData(&gps_data, M17_GNSS_STATION_HANDHELD);
+        type.fields.encType    = M17_ENCRYPTION_NONE;
```

Comment:
I'd move the setting of the LSF type field inside the `setGnssData` function itself: being it a member function of the M17LinkSetupFrame class, it has access to all the fields of the LSF data structure.